openupgradelib/openupgrade.py

```diff
@@ -2011,3 +2020,100 @@ def chunked(records, single=True):
+                env.cr,
+                """ UPDATE ir_model_data
+                    SET res_id = %s WHERE id = %s """, (new_id, data[0]))
+
+    if ('ir_translation', 'res_id') not in exclude_columns:
+        logged_query(
+            env.cr,
+            """ UPDATE ir_translation SET res_id = %(new_id)s
+                WHERE type = 'model' AND res_id = %(old_id)s
+                AND name like %(model)s || ',%%'""",
+            values)
+
+    if ('ir_property', 'value_reference') not in exclude_columns:
+        # Handle properties that reference to this model
+        logged_query(
+            env.cr,
+            """ UPDATE ir_property
+                SET value_reference = %(model)s || ',' || %(new_id)s
+                WHERE value_reference = %(model)s || ',' || %(old_id)s""",
+            values)
+
+    # Tables with generic model/res_id references
+    for table, res_id_column, model_column in [
+            ('calendar_event', 'res_id', 'res_model'),
+            ('ir_attachment', 'res_id', 'res_model'),
+            ('mail_activity', 'res_id', 'res_model'),
+            ('mail_followers', 'res_id', 'model'),
+            ('mail_message', 'res_id', 'model'),
+            ('rating_rating', 'res_id', 'res_model'),
+            ]:
+        if (table, res_id_column) in exclude_columns:
+            continue
+        if table_exists(env.cr, table):
+            logged_query(
+                env.cr,
+                """ UPDATE {table} SET {res_id_column} = %(new_id)s
+                    WHERE {model_column} = %(model)s
+                        AND {res_id_column} = %(old_id)s
+                """.format(
+                    table=AsIs(table), res_id_column=AsIs(res_id_column),
+                    model_column=AsIs(model_column)),
+                values)
+
+    # https://stackoverflow.com/questions/1152260
+    # /postgres-sql-to-list-table-foreign-keys
+    env.cr.execute(
+        """ SELECT tc.table_name, kcu.column_name
+            FROM information_schema.table_constraints AS tc
+            JOIN information_schema.key_column_usage AS kcu
+                ON tc.constraint_name = kcu.constraint_name
+                AND tc.table_schema = kcu.table_schema
+            JOIN information_schema.constraint_column_usage AS ccu
+                ON ccu.constraint_name = tc.constraint_name
+                AND ccu.table_schema = tc.table_schema
+            WHERE constraint_type = 'FOREIGN KEY'
+            AND ccu.table_name = %s and ccu.column_name = 'id'
+        """, (env[model]._table,))
+    for table, column in env.cr.fetchall():
+        if (table, column) in exclude_columns:
+            continue
+        query = env.cr.mogrify(
+            """ UPDATE {table} SET {column} = %(new_id)s
+                WHERE {column} = %(old_id)s""".format(
+                    table=AsIs(table), column=AsIs(column)), values)
+        with allow_pgcodes(env.cr, errorcodes.UNIQUE_VIOLATION, query=query):
+            logged_query(env.cr, query)
```

Comment:
If `logged_query` already logs it, why do we need the `query` kwarg in `allow_pgcodes`? :thinking:

Reply:
Logged query logs afterwards, to mention the number of affected records. If an exception occurs, the offending query is not logged. As I'm writing this, it occurs to me that I should solve this in logged_query instead....